runtime-testsuite/test/org/antlr/v4/test/runtime/java/BaseJavaTest.java

```diff
@@ -461,7 +461,7 @@ public void testActions(String templates, String actionName, String action, Stri
             AnalysisPipeline anal = new AnalysisPipeline(g);
             anal.process();
 
-			CodeGenerator gen = new CodeGenerator(g);
+			CodeGenerator gen = CodeGenerator.createCodeGenerator(g);
```

Comment:
just curious why you converted this to a factory method.  again I'm looking for gratuitous changes ;)

Reply:
Because it returns `null` if `CodeGenerator` can not be created (and throws error `CANNOT_CREATE_TARGET_GENERATOR`). A contructor can not return not null value. Also, I removed duplication of some logic in `CodeGnerator`.

Reply:
cool. thanks.

Reply:
Maybe it makes sense to rename to `CodeGenerator.create` instead of `CodeGenerator.createCodeGenerator` (a bit redundancy).

Reply:
oh. yeah.  that makes sense.